lib/components/transaction-sponsor-wallet.ts

```diff
@@ -0,0 +1,46 @@
+import { Wallet } from "ethers";
+import { Address, SupportedProvider } from "../types" ;
+import { waitAndReturn } from "../util";
+import TransactionSponsor from "./transaction-sponsor";
+
+export default class TransactionSponsorWallet extends TransactionSponsor {
+
+    private funder: Wallet;
+
+    constructor(
+        provider: SupportedProvider,
+        funderPrivateKey: Address,
+        fundingAmount: bigint,
+        confirmations: number | undefined = undefined
+    ) {
+        super(fundingAmount, confirmations);
+        this.funder = new Wallet(funderPrivateKey, provider);
+    }
+
+    withFunder(newFunder: Wallet) {
+        this.setFunder(newFunder) ;
+        return this ;
+    }
+
+    setFunder(newFunder: Wallet) {
+        this.funder = newFunder ;
+    }
```

Comment:
I think this method could be private, as is already publicly used by the fluent interface